ethereum/migrations/2_next.js

```diff
@@ -63,5 +64,30 @@ module.exports = function(deployer, network, accounts) {
     );
 
     await deployer.deploy(TestToken, 100000000, "Test Token", "TEST");
+
+    // Deploy ERC1820 Registry for our E2E stack.
+    if (network == 'e2e_test')  {
+
+      require('@openzeppelin/test-helpers/configure')({ web3 });
+      const { singletons } = require('@openzeppelin/test-helpers');
+
+      await singletons.ERC1820Registry(accounts[0]);
+    }
+
+    // only deploy this contract to non-development networks
```

Comment:
why?

Reply:
Because these migration scripts interfere with unit tests (which use the development network). 

Truffle assumes that the unit tests need the contracts deployed, so it runs this migration script before running tests. But actually most of the unit tests need to redeploy the contracts before each individual test in a `beforeEach` function.

So there's this duplicated deployment which has the potential to introduce deployment errors, because the migration script creates state that is incompatible with what the unit tests are doing (or vice versa).

This has worked up until now without errors. But now we've introduced DOTApp and its indirect dependency on an ERC1820 registry: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC777/ERC777.sol#L31.

The only way I've got it to work is to deploy the ERC1820 registry in the unit tests.